### PR TITLE
Added support of `pallet-assets` and chain extension for it.

### PR DESCRIPTION
**Pull request by xgreenx**
Added support of the [chain extension](https://github.com/Supercolony-net/pallet-assets-chain-extension) for `pallet-assets`. The extension requires additional changes on the `pallet-assets` side to be ready for production. But we ca enable it already for testing=)

### CODE REVIEW

File: runtime/Cargo.toml [hunk 1]
```diff
@@ -51,6 +51,10 @@ pallet-contracts = { git = "https://github.com/paritytech/substrate", package =
 pallet-contracts-primitives = { git = "https://github.com/paritytech/substrate", package = "pallet-contracts-primitives", default-features = false }
 pallet-contracts-rpc-runtime-api = { git = "https://github.com/paritytech/substrate", package = "pallet-contracts-rpc-runtime-api", default-features = false }
 
+# Chain extension
+pallet-assets = { git = "https://github.com/paritytech/substrate", package = "pallet-assets", default-features = false }
```
Comment (athei) [inline runtime/Cargo.toml:55]: I think you forgot to propagate the `runtime-benchmarks` feature.

**Review by athei — COMMENTED** [review]
Something went wrong with fetching from your git repo.

File: runtime/Cargo.toml (hunk 1 above)
Comment (xgreenx) [reply runtime/Cargo.toml:55]: I used the wrong hash commit=)

Comment (athei) [reply runtime/Cargo.toml:55]: > I think you forgot to propagate the `runtime-benchmarks` feature
👆

**Review by athei — APPROVED** [review]
I am a bit worried that this will make our update process more complicated because we are restrained to the substrate version your repo uses. But I see no better way here. We don't want a node per extension.

**Comment by HCastano** [comment]
> I am a bit worried that this will make our update process more complicated because we are restrained to the substrate version your repo uses. But I see no better way here. We don't want a node per extension.

Maybe this is a good time for us to start pinning to release branches instead of following `master`. Green can do the same in the `obce` repo and we can more easily stay in lockstep there

**Comment by athei** [comment]
If you update your extension to use the `polkadot-v0.9.29` branch we can merge this.

File: runtime/Cargo.toml [hunk 2]
```diff
@@ -51,6 +51,10 @@ pallet-contracts = { git = "https://github.com/paritytech/substrate", package =
 pallet-contracts-primitives = { git = "https://github.com/paritytech/substrate", package = "pallet-contracts-primitives", default-features = false , branch = "polkadot-v0.9.29" }
 pallet-contracts-rpc-runtime-api = { git = "https://github.com/paritytech/substrate", package = "pallet-contracts-rpc-runtime-api", default-features = false , branch = "polkadot-v0.9.29" }
 
+# Chain extension
+pallet-assets = { git = "https://github.com/paritytech/substrate", package = "pallet-assets", default-features = false, branch = "polkadot-v0.9.29" }
```
Comment (athei) [inline runtime/Cargo.toml:55]: Can you move that up to the other pallets? This is not a chain extension per se.

**Pull request merged**